File: jimmer-ddd-java/src/main/java/org/morecup/jimmerddd/core/sqlclient/AggregationEntityDraft.java
package org.morecup.jimmerddd.core.sqlclient;

import org.babyfish.jimmer.Draft;
import org.babyfish.jimmer.internal.GeneratedBy;
import org.babyfish.jimmer.lang.OldChain;
import org.babyfish.jimmer.meta.ImmutablePropCategory;
import org.babyfish.jimmer.meta.ImmutableType;

import java.util.Collections;

@GeneratedBy(
        type = AggregationEntity.class
)
public interface AggregationEntityDraft extends AggregationEntity, Draft {
    Producer $ = Producer.INSTANCE;

    @OldChain
    AggregationEntityDraft setIdPreLoaded(Boolean idPreLoaded);

    @GeneratedBy(
            type = AggregationEntity.class
    )
    class Producer {
        static final Producer INSTANCE = new Producer();

        public static final ImmutableType TYPE = ImmutableType
            .newBuilder(
                "0.9.81",
                AggregationEntity.class,
                Collections.emptyList(),
                null
            )
            .add(-1, "idPreLoaded", ImmutablePropCategory.SCALAR, Boolean.class, true)
            .build();

        private Producer() {
        }
    }
}

File: jimmer-ddd-kotlin/src/main/kotlin/org/morecup/jimmerddd/core/sqlclient/AggregationEntityDraft.kt
@file:Suppress("warnings")

package org.morecup.jimmerddd.core.sqlclient

import kotlin.Boolean
import kotlin.Suppress
import org.babyfish.jimmer.Draft
import org.babyfish.jimmer.`internal`.GeneratedBy
import org.babyfish.jimmer.kt.DslScope
import org.babyfish.jimmer.meta.ImmutablePropCategory
import org.babyfish.jimmer.meta.ImmutableType

@DslScope
@GeneratedBy(type = AggregationEntity::class)
public interface AggregationEntityDraft : AggregationEntity, Draft {
    override var idPreLoaded: Boolean

    @GeneratedBy(type = AggregationEntity::class)
    public object `$` {
        public val type: ImmutableType = ImmutableType
                    .newBuilder(
                        "0.9.81",
                        AggregationEntity::class,
                        listOf(

                        ),
                        null
                    )
                    .add(-1, "idPreLoaded", ImmutablePropCategory.SCALAR, Boolean::class.java, false)
                    .build()
    }
}
